### PR TITLE
5. Add support for native (database) Ids

### PR DESCRIPTION
**Pull request by Andy2003**
merge after #51

### CODE REVIEW

File: readme.adoc [hunk 1]
```diff
@@ -338,6 +338,46 @@ The default Neo4j types are handled both as argument types as well as field type
 
 NOTE: Datetime and spatial not yet.
 
+=== Usage of ID
```
Comment (jexp) [inline readme.adoc]: later: move to end

File: src/main/kotlin/org/neo4j/graphql/GraphQLExtensions.kt [hunk 2]
```diff
@@ -153,6 +154,8 @@ fun paramName(variable: String, argName: String, value: Any?): String = when (va
 }
 
 fun FieldDefinition.isID(): Boolean = this.type.name() == "ID"
+fun FieldDefinition.isNativeId() = this.name == ProjectionBase.NATIVE_ID
```
Comment (jexp) [inline src/main/kotlin/org/neo4j/graphql/GraphQLExtensions.kt]: later: perhaps via directive

File: src/main/kotlin/org/neo4j/graphql/handler/BaseDataFetcher.kt [hunk 3]
```diff
@@ -102,16 +102,25 @@ abstract class BaseDataFetcher(
...
+                isRelation: Boolean,
                 paramName: String? = idProperty?.let { paramName(variable, idProperty.name, idProperty.value) }
         ): Cypher {
             return when {
                 idProperty != null && paramName != null -> {
                     val queryParams = mapOf(paramName to idProperty.value.toJavaValue())
-                    // TODO handle @property aliasing
-                    if (idProperty.value is ArrayValue) {
-                        Cypher("($variable:$label) WHERE  $variable.${idField.name.quote()} IN $$paramName", queryParams)
+                    if (idField.isNativeId()) {
```
Comment (jexp) [inline src/main/kotlin/org/neo4j/graphql/handler/BaseDataFetcher.kt]: lots of nested conditionals, need to find a nicer way.

File: src/main/kotlin/org/neo4j/graphql/handler/BaseDataFetcher.kt [hunk 4]
```diff
@@ -102,16 +102,25 @@ abstract class BaseDataFetcher(
...
+                        if (isRelation) {
+                            Cypher("()-[$variable:$label]->() WHERE ID($variable) = $$paramName", queryParams)
+                        } else {
+                            Cypher("($variable:$label) WHERE ID($variable) = $$paramName", queryParams)
+                        }
                     } else {
-                        Cypher("($variable:$label { ${idField.name.quote()}: $$paramName })", queryParams)
+                        // TODO handle @property aliasing
+                        if (idProperty.value is ArrayValue) {
```
Comment (jexp) [inline src/main/kotlin/org/neo4j/graphql/handler/BaseDataFetcher.kt]: better check?

File: src/main/kotlin/org/neo4j/graphql/handler/CreateTypeHandler.kt [hunk 5]
```diff
@@ -20,7 +17,7 @@ class CreateTypeHandler private constructor(
             if (relevantFields.isEmpty()) {
                 return null
             }
-            val fieldDefinition = createFieldDefinition("create", type.name(), relevantFields, false).build()
+            val fieldDefinition = createFieldDefinition("create", type.name(), relevantFields.filter { !it.isNativeId() }, false).build()
```
Comment (jexp) [inline src/main/kotlin/org/neo4j/graphql/handler/CreateTypeHandler.kt]: false -> use param name

Comment (Andy2003) [reply src/main/kotlin/org/neo4j/graphql/handler/CreateTypeHandler.kt]: comes in the next PR ;-)

File: src/main/kotlin/org/neo4j/graphql/handler/relation/BaseRelationHandler.kt [hunk 6]
```diff
@@ -81,7 +81,7 @@ abstract class BaseRelationHandler(
             throw java.lang.IllegalArgumentException("No ID for the ${if (start) "start" else "end"} Type provided, ${idField.argumentName} is required")
         }
         return getSelectQuery(relFieldName, idField.declaringType.label(), arguments[idField.argumentName],
-                idField.field, (relFieldName + idField.argumentName.capitalize()).quote())
+                idField.field, false, (relFieldName + idField.argumentName.capitalize()).quote())
```
Comment (jexp) [inline src/main/kotlin/org/neo4j/graphql/handler/relation/BaseRelationHandler.kt]: false -> add param name

File: src/test/resources/augmentation-tests.adoc [hunk 7]
```diff
@@ -786,7 +1028,7 @@ type Person4 {
 
 type Person5 implements HasMovies {
   id: ID!
-  movies(first: Int, offset: Int): [Movie]
+  movies(first: Int, offset: Int): [Movie] @relation(direction : OUT, from : "from", name : "LIKES", to : "to")
```
Comment (jexp) [inline src/test/resources/augmentation-tests.adoc]: from and to are defaults, OUT is also default

Comment (Andy2003) [reply src/test/resources/augmentation-tests.adoc]: this is just copied from what the SchemaWriter prints out

**Pull request merged**